### PR TITLE
ZAYIN Buffs

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

- You Are Bald and its EGO now checks for trait instead of user's hairstyle.
- You Are Bald heals sanity of bald people that work on it(up to a total of 50% sanity).
- The bottle of tears debuff now lasts for 5 minutes, instead of 10.
- One sin's sanity heal is buffed from static 30 to 50%; For insane people it's 25% instead.
- Fixed/Improved some code things.

## Why It's Good For The Game

- Previously you could cheese the system by using mirror. NOT ANYMORE!!!
- Adds a reason to work on You Are Bald. It's not a very good reason, as one sin does the job better, but it's still some use for ZAYIN late in the game.
- 10 minutes, in my opinion, is too long for what is effectively meant to be a good thing(a way to pass some reverse stat checks).
- One sin is more useful now.
- Mostly fixes some code that wasn't done in the proper "style" and some text tweaks.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm [hunk 1]
```diff
@@ -25,37 +25,40 @@
...
 
 /mob/living/simple_animal/hostile/abnormality/bald/WorkChance(mob/living/carbon/human/user, chance)
-	if(user.hairstyle in balding_list)
+	if(HAS_TRAIT(user, TRAIT_BALD))
 		return 95
 	return chance
 
+/mob/living/simple_animal/hostile/abnormality/bald/WorktickSuccess(mob/living/carbon/human/user)
+	if(HAS_TRAIT(user, TRAIT_BALD))
+		user.adjustSanityLoss(user.maxSanity * 0.05) // Half of sanity restored for bald people
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm:38]: This is 5% not 50% if I'm not mistaken.

File: code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm [hunk 2]
```diff
@@ -25,37 +25,40 @@
 	gift_chance = 10
 	gift_message = "Now we're feeling awesome!"
 
-	var/list/balding_list = list(
-							"Bald"
-							)
+	var/list/balding_list = list("Bald")
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm:28]: From what I can see, this is now redundant and unused.

File: code/modules/mob/living/simple_animal/abnormality/zayin/one_sin.dm [hunk 3]
```diff
@@ -66,14 +66,16 @@
...
 		for(var/mob/living/carbon/human/H in GLOB.player_list)
-			H.adjustSanityLoss(30)
-	..()
-
+			if(H.z != z)
+				continue
+			var/heal_factor = 0.5
+			if(H.sanity_lost)
+				heal_factor = 0.25
+			H.adjustSanityLoss(H.maxSanity * heal_factor)
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/one_sin.dm:80]: This does not exclude the worker, meaning they'd be healed twice and probably fully heal their sanity or at least 75% if somehow insane at max PE.

File: code/modules/projectiles/projectile/ego_bullets/zayin.dm [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /obj/projectile/ego_bullet/ego_tough
 	name = "9mm tough bullet"
-	damage = 8
+	damage = 10 // Being bald is the optimal gameplay choice!
```
Comment (dsmith328) [inline code/modules/projectiles/projectile/ego_bullets/zayin.dm:3]: image

**Review by dsmith328 — CHANGES_REQUESTED** [review]
Mainly those 3 should be fixed before being released.

File: code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm:38]: > This is 5% not 50% if I'm not mistaken.

It's 5% per success tick, which we'll be having 10 of.

File: code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/zayin/bald.dm:28]: Will remove.

File: code/modules/mob/living/simple_animal/abnormality/zayin/one_sin.dm (hunk 3 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/zayin/one_sin.dm:80]: I think it's not an issue, personally.

**Review by dsmith328 — APPROVED** [review]
Optimal Zayin Behavior

**Pull request merged**